hotel_realta.sql

```diff
@@ -15,14 +15,96 @@ GO
 CREATE SCHEMA Users;
 GO
 
--- dummy schema master
 CREATE SCHEMA Master;
 GO
 
--- dummy table master.members
-CREATE TABLE master.members (
-	memb_name NVARCHAR(15) NOT NULL,
-	CONSTRAINT pk_memb_name PRIMARY KEY(memb_name)
+CREATE TABLE Master.regions (
+  region_code int IDENTITY(1, 1),
+  region_name nvarchar(35) UNIQUE NOT NULL,
+  CONSTRAINT pk_region_code PRIMARY KEY(region_code)
+);
+
+CREATE TABLE Master.country (
+  country_id int IDENTITY(1, 1),
+  country_name nvarchar(55) UNIQUE NOT NULL,
+  country_region_id int,
+  CONSTRAINT pk_country_id PRIMARY KEY (country_id),
+  CONSTRAINT fk_country_region_id FOREIGN KEY(country_region_id) REFERENCES Master.regions(region_code) 
+    ON DELETE CASCADE 
+    ON UPDATE CASCADE
+);
+
+CREATE TABLE Master.provinces (
+  prov_id int IDENTITY (1, 1),
+  prov_name nvarchar(85) NOT NULL,
+  prov_country_id int CONSTRAINT pk_prov_id PRIMARY KEY(prov_id),
+  CONSTRAINT fk_prov_country_name FOREIGN KEY(prov_country_id) REFERENCES Master.country(country_id) 
+    ON DELETE CASCADE 
+    ON UPDATE CASCADE
+);
+
+CREATE TABLE Master.address (
+  addr_id int IDENTITY(1, 1),
+  addr_line1 nvarchar(255) NOT NULL,
+  addr_line2 nvarchar(255),
+  addr_postal_code nvarchar(5),
+  addr_spatial_location geography,
+  addr_prov_id int,
+  CONSTRAINT pk_addr_id PRIMARY KEY(addr_id),
+  CONSTRAINT fk_addr_prov_id FOREIGN KEY(addr_prov_id) REFERENCES Master.provinces(prov_id) 
+    ON DELETE CASCADE 
+    ON UPDATE CASCADE
+);
+
+CREATE TABLE Master.category_group (
+  cagro_id int IDENTITY(1, 1),
+  cagro_name nvarchar(25) UNIQUE NOT NULL CHECK (cagro_name IN ('1', '2', '3 ', '4', '5', '6', '7')),
```

Comment:
cagro_name nvarchar(25) UNIQUE NOT NULL CHECK (cagro_name IN ('Room', 'Restaurant', 'Meeting Room', 'Gym', 'Aula', 'Swimming Pool', 'Balroom'))